Assets/Scripts/PlayerShoot.cs

```diff
@@ -43,40 +43,33 @@ private void Update()
         if (PlayerUI.isPaused)
             return;
 
-        if (Input.GetButtonDown("Fire1")) // if we click on the left mouse button
+        if (Input.GetButtonDown("Fire1") && inputManager.Aiming) // if we click on the left mouse button
         {
             Shoot();
         }
     }
 
     private void Shoot()
     {
-        RaycastHit hit;
-        if (Physics.Raycast(cam.transform.position, cam.transform.forward, out hit, weapon.range, mask))
+        if (!Physics.Raycast(cam.transform.position, cam.transform.forward, out var hit, weapon.range, mask)) return;
+        
+        if(!hit.collider.CompareTag("Product"))
         {
-            if(!hit.collider.CompareTag("Product"))
-            {
-                Debug.Log("This is not a scannable product");
-                return;
-            }
-
-            if(!inputManager.Aiming)
-            {
-                return;
-            }
-
-            var hitName = hit.collider.name;
+            Debug.Log("This is not a scannable product");
+            return;
+        }
+        
+        var hitName = hit.collider.name;
 
-            StoreItem storeItem = hit.collider.gameObject.GetComponent<StoreItem>();
-            Debug.Log("You hitted " + storeItem.displayedName);
-            ScanningAudioSource.Play();
+        StoreItem storeItem = hit.collider.gameObject.GetComponent<StoreItem>();
+        Debug.Log("You hitted " + storeItem.displayedName);
+        ScanningAudioSource.Play();
 
-            bool scanResult = playerScanController.Scan(storeItem.displayedName);
+        var scanResult = playerScanController.Scan(storeItem.displayedName);

```

Comment:
If the Scan function return a bool, why changing this type from bool to var ? 
if you want to keep "var" because it's easier, give a better name to the variable (eg. isScanValid)
```suggestion
        var isScanValid = playerScanController.Scan(storeItem.displayedName);
```